cmd/state-svc/internal/server/server.go

```diff
@@ -32,13 +33,13 @@ type Server struct {
 	analytics   *sync.Client
 }
 
-func New(cfg *config.Instance, an *sync.Client, shutdown context.CancelFunc) (*Server, error) {
+func New(cfg *config.Instance, an *sync.Client, checker *deprecation.Checker, shutdown context.CancelFunc) (*Server, error) {
```

Comment:
I don't think we need to pass checker into the server. The checker is pretty much confined to the resolver package, so constructing it can happen there.